AdminUi/apps/admin_ui/lib/pages/login_page.dart

```diff
@@ -34,56 +38,128 @@ class _LoginScreenState extends State<LoginScreen> {
+                            ),
+                            focusedErrorBorder: const OutlineInputBorder(
+                              borderSide: BorderSide(color: Colors.red, width: 2),
+                            ),
+                            errorText: (_attemptedLogin && !apiKeyValid) ? 'Invalid API Key' : null,
+                          ),
+                          onSubmitted: (_) => _login(),
+                          onChanged: (text) {
+                            setState(() {
+                              _isButtonEnabled = text.isNotEmpty;
+                              _attemptedLogin = false;
+                              apiKeyValid = false;
+                            });
+                          },
+                          obscureText: true,
+                        ),
+                        const SizedBox(height: 20),
+                        SizedBox(
+                          width: double.infinity,
+                          height: 40,
+                          child: ElevatedButton(
+                            onPressed: _isButtonEnabled ? _login : null,
+                            style: _isButtonEnabled
+                                ? const ButtonStyle(
+                                    backgroundColor: MaterialStatePropertyAll(
+                                      Color.fromARGB(255, 0, 47, 114),
+                                    ),
+                                  )
+                                : null,
+                            child: Text(
+                              'Login',
+                              style: TextStyle(color: _isButtonEnabled ? Colors.white : null),
+                            ),
+                          ),
+                        ),
+                        const SizedBox(height: 8),
+                        TextButton(
+                          onPressed: () {
+                            // TODO(stamenione): Implement what should happen when "Forgot API Key?" is pressed
+                          },
+                          child: const Text(
+                            'Forgot API Key?',
+                            style: TextStyle(
+                              color: Color.fromARGB(255, 0, 47, 114),
+                            ),
+                          ),
+                        ),
+                      ],
+                    ),
+                  ),
                 ),
-                onSubmitted: (_) => _login(),
-                obscureText: true,
-              ),
-              const SizedBox(height: 20),
-              ElevatedButton(
-                onPressed: _login,
-                child: const Text('Login'),
               ),
-            ],
+            ),
           ),
         ),
       ),
+      backgroundColor: const Color.fromARGB(255, 211, 228, 255),
     );
   }
 
   Future<void> _login() async {
+    _attemptedLogin = true;
     final apiKey = _apiKeyController.text.trim();
```

Comment:
What if the API key actually contains a whitespace at the start or the end? In this case the user will never be able to login.

Reply:
I didn't think about this. Is that possible?

Reply:
At least the backbone doesn't have any kind of validation on the api key so far. 

Counter question: what sense does it make to have the trim? :)

Reply:
IDK